### PR TITLE
CORTX-29705:RGW:updation of rgw service name as rgw -> rgw_s3 in motr client & --services parameters

### PR DESCRIPTION
**Pull request by sachitanands**
Signed-off-by: Sachitanand Shelake <sachitanand.shelake@seagate.com>

Refine Config Structure for RGW

MOTR clients i.e. using service names as client names, e.g. "motr_client" "rgw_s3"

RGW Mini Provisioner to be verified with "rgw_setup --service=rgw_s3" 

Reference doc : https://seagate-systems.atlassian.net/wiki/spaces/PUB/pages/611615649/CORTX+Configuration+Parameters?src=mail&src.mail.action=view&src.mail.notification=com.atlassian.confluence.plugins.confluence-notifications-batch-plugin%3Abatching-notification&src.mail.recipient=8a7f808a7812f7f70178188db13e05bf&src.mail.timestamp=1648099792383 

# Problem Statement
- Problem statement

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Review by rahul-tripathi-git — APPROVED** [review]
LGTM

**Review by shriya-deshmukh — APPROVED** [review]

**Comment by sachitanands** [comment]
Trying deployment using below job,
build : https://eos-jenkins.colo.seagate.com/job/GitHub-custom-ci-builds/job/generic/job/custom-ci/5472/console 
docker image : 
      cortx-docker.colo.seagate.com/seagate/cortx-all:2.0.0-5472-custom-ci 
      cortx-docker.colo.seagate.com/seagate/cortx-rgw:2.0.0-5472-custom-ci
 deployment job : https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/RGW/job/setup-cortx-rgw-cluster/1506

**Review by cdeshmukh — APPROVED** [review]

**Comment by cdeshmukh** [comment]
Comment from Ujjwal

> You dont need to depend on provisioner team. Here is the steps

Install cortx-rgw rpm

Create gconf yaml sample file.

Call rgw_setup with parameter

No need of K8s or provisioner.

File: src/rgw/const.py [hunk 1]
```diff
@@ -17,6 +17,10 @@
 from enum import Enum
 
 COMPONENT_NAME = 'rgw'
+COMPONENT_SVC_NAME = 'rgw_s3'
+# service name be passed as --service parameter
+# e.g. rgw_setup --service={rgw_s3|all}
+SUPPORTED_SERVICE_NAMES = [COMPONENT_SVC_NAME, 'all']
```
Comment (ujjwalpl) [inline src/rgw/const.py:23]: Remove this from this place. This variable can be part of individual code.

**Review by ujjwalpl — CHANGES_REQUESTED** [review]

File: src/rgw/const.py (hunk 1 above)
Comment (sachitanands) [reply src/rgw/const.py]: done

**Comment by sachitanands** [comment]
Triggered deployment job with final changes,
1) docker images : 
https://eos-jenkins.colo.seagate.com/job/GitHub-custom-ci-builds/job/generic/job/custom-ci/5482/console 
  cortx-docker.colo.seagate.com/seagate/cortx-all:2.0.0-5482-custom-ci 
  cortx-docker.colo.seagate.com/seagate/cortx-rgw:2.0.0-5482-custom-ci
 2) Deployment job : https://eos-jenkins.colo.seagate.com/job/Cortx-Automation/job/RGW/job/setup-cortx-rgw-cluster/1547/  (In-progress)
 
 meanwhile will do manual testing as mentioned by Chetan in earlier comments.

**Comment by rahul-tripathi-git** [comment]
MicrosoftTeams-image (3)

@sachitanands tested the manual triggering of mini provisioning, followed following steps:
1. triggered post install phase with a wrong service name, abcd, failed as expected.
2. triggered post install phase with 'all' as service name, passed as expected.
3. triggered post install phase with rgw_s3 as service name, passed as expected.

**Comment by cdeshmukh** [comment]
thanks @rahul-tripathi-git. Just note we cant merge this yet until changes reflected in GConf from Provisioner. We just wanted to confirm if these changes are needed for sure, or existing code would have worked.

**Review by ujjwalpl — APPROVED** [review]

**Comment by cdeshmukh** [comment]
In morning, @sachitanands @rahul-tripathi-git confirmed that deployment works with or without these changes, that way we dont need to wait for GConf or even Hare changes.  Though these PR changes are better to align with services argument consumption.  @ujjwalpl suggested too to go ahead with merge in main branch

**Pull request merged**